### DISCUSSION TITLE
Toggle Dark Mode

### DISCUSSION THREAD

**Discussion by jeffchown** [livewire/flux#384] [Feature Requests] [answered]
Morning, @calebporzio and team.

Could you share the code you use on https://fluxui.dev for the Toggle Dark Mode function?
I'd like to add it to an existing app during development so I can easily view/test both modes of my newly Fluxified pages :)

Thanks!

**Comment by ITSYV** [comment, marked as answer]
@jeffchown One way is to add this to your tailwind conf:
`module.exports = {
    darkMode: 'selector',
    ...
}`
See: https://fluxui.dev/docs/installation#using-dark-mode

After adding this, you can toggle a "dark" class on your html or body tag with Alpine.
At the FluxUI website they store a boolean with javascript in "darkMode" ("x-on:click="$store.darkMode.toggle()"). But you can do the same storing a session or cookie and retrieving this in your app layout file 


For local dev, you could just add / remove "dark" to your html/body tags to switch between themes.

**Reply by jeffchown** [reply]
Thanks for the quick reply @ITSYV 

I have implemented something similar to what you suggest, but when I've toggled dark mode on, each time I navigate to another page in my app, it blinks light mode quickly, then re-renders in dark mode.

I use this script in the header of each page:
```
<script>
    document.addEventListener('alpine:init', () => {
        Alpine.store('darkMode', {
            on: Alpine.$persist(false).as('darkMode_on'),

            init() {
                if (this.on === 'undefined') {
                    console.log('undefined')
                    this.on = window.matchMedia('(prefers-color-scheme: dark)').matches
                }

                this.updateDarkClass();
            },


            updateDarkClass() {
                if (this.on) {
                    document.documentElement.classList.add('dark');
                } else {
                    document.documentElement.classList.remove('dark');
                }
            },

            toggle() {
                this.on = ! this.on
                console.log('darkMode.toggle()', this.on);

                this.updateDarkClass();
            }
        })
    })
</script>
```

Then, this for my toggle icon:

```
<flux:navbar.item
    wire:navigate
    x-data
    x-on:click.prevent="$store.darkMode.toggle()"
    id="theme-toggle"
    class="max-lg:hidden"
    href="#"
    label="Toggle Dark Mode"
>
    <div class="w-5">
        <flux:icon.moon class="size-5" x-show="! $store.darkMode.on" />
        <flux:icon.moon variant="solid" class="size-5" x-show="$store.darkMode.on" x-cloak />
    </div>
</flux:navbar.item>
```

**Reply by ITSYV** [reply]
Are you working SPA style or just navigating the old-school way?
One solution could be to use wire:navigate so that css files aren't reloaded on navigating?

Not sure how to prevent this in another way

**Reply by jeffchown** [reply]
After removing the `wire:navigate` and `href='#'` from my `flux:navbar.item` it helped eliminate the page refresh on toggle, but I still get the brief light mode 'flash' then dark mode re-render when navigating to pages without using `wire:navigate`.

**Reply by jeffchown** [reply]
> Are you working SPA style or just navigating the old-school way? One solution could be to use wire:navigate so that css files aren't reloaded on navigating?
> 
> Not sure how to prevent this in another way

We were typing at the same time :). Most of my link use `wire:navigate` and they perform without the 'blink', it is only the legacy links (that I still need to convert/test using `wire:navigate`) that cause the blinking.

**Reply by jeffchown** [reply]
I think you may be right re: no way to prevent this on the non-`wire:navigate` links/navigation

**Reply by ITSYV** [reply]
Hmm, I guess some more experienced javascript dev should answer this.
Maybe it's because Alpine isn't loaded simultaniously? (because your code runs after Alpine initializes) 
Does it work if you set it in javascript in your head tag?

**Reply by jeffchown** [reply]
If you mean via raw js in my head tag, haven't tried that yet.

I'll probably `wire:navigate` all my links first, then maybe revist that afterwards as I am curious to see if there's a way to make it work for all links without the blink.

**Reply by jeffchown** [reply]
Thanks again for your help, @ITSYV 🙏🏻
